scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,28 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+set +e
+dpkg-query -l opensearch | grep ^ii
+OPENSEARCH_INSTALLED=$? # 0 when installed, 1 when installed
```

Comment:
Missing negation :wink:

```suggestion
OPENSEARCH_INSTALLED=$? # 0 when installed, 1 when not installed
```

But this make the `OPENSEARCH_INSTALLED` meaning misleading (at least for me). What about using yes/no:

```suggestion
if [ $? -eq 0 ]; then
  OPENSEARCH_INSTALLED=yes
else
  OPENSEARCH_INSTALLED=no
fi
```

This makes it more readable after IMHO.